Sources/TuistCore/Graph/GraphLoader.swift

```diff
@@ -215,9 +236,27 @@ public final class GraphLoader: GraphLoading {
         case .xctest:
             return try loadXCTestSDK(platform: fromPlatform)
 
-        case .thirdParty:
-            // A dependency imported through Dependencies.swift.
-            fatalError("TargetDependency.thirdParty not implemented yet")
+        case let .thirdParty(name):
```

Comment:
Would this be something that can be modelled via a mapper instead of extending the `GraphLoader`?

e.g.

```swift
class ThirdPartyDependencyMapper: TargetMapping {
    let dependencies: DependenciesGraph
    func map(target: Target) throws -> (Target, [SideEffectDescriptor]) {
        // convert TargetDepdendency.thirdParty to appropriate equivalent
    }
}
```

Reply:
What about removing the `TuistGraph.TargetDependency.thirdParty` case and do the mapping in the `TargetDependency+ManifestMapper`, mapping from `ProjectDefinition.TargetDependency` to the corresponding `TuistGraph.TargetDependency` (i.e. either xcframework or project)?

Reply:
> What about removing the TuistGraph.TargetDependency.thirdParty case and do the mapping in the TargetDependency+ManifestMapper, mapping from ProjectDefinition.TargetDependency to the corresponding TuistGraph.TargetDependency (i.e. either xcframework or project)?

That's also an option, that's similar to how Plugins by name are integrated for resource synthesisers

Reply:
I have updated the logic, let me know what you think 🙌